plugins/device/deviceutils/store-utils.go

```diff
@@ -51,14 +52,52 @@ func GetDevicesFromPeer(peerInfo *peer.Peer) ([]deviceapi.Info, error) {
 	return deviceInfo, nil
 }
 
+// SetDeviceState sets device state and updates device state in etcd
+func SetDeviceState(peerID string, deviceName string, deviceState string) error {
+
+	devices, err := GetDevices(peerID)
+	if err != nil {
+		return err
+	}
+
+	index, exists := DeviceInList(deviceName, devices)
+	if exists {
+		devices[index].State = deviceState
+		err := UpdateDevices(peerID, devices)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New("Device doesnot exists in the given peer")
+	}
+	return nil
+}
+
+func UpdateDevices(peerID string, devices []deviceapi.Info) error {
+	peerInfo, err := peer.GetPeer(peerID)
+	if err != nil {
+		return err
+	}
+	deviceJSON, err := json.Marshal(devices)
+	if err != nil {
+		return err
+	}
+	peerInfo.Metadata["_devices"] = string(deviceJSON)
+	err = peer.AddOrUpdatePeer(peerInfo)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 // DeviceInList checks whether the given device is in list of devices or not.
-func DeviceInList(reqDevice string, devices []deviceapi.Info) bool {
-	for _, key := range devices {
+func DeviceInList(reqDevice string, devices []deviceapi.Info) (int, bool) {
```

Comment:
If the device is not found you are setting the index to ```-1```. an index can be used to verify whether the device is present or not. IMO no need to have one more variable.

Reply:
@Madhu-1  if we want to edit a particular device from the list if the device exists, then returning the index is useful.

Reply:
we can use the index and remove bool value. to check whether the device is in a list, you can make use of index variable itself.